tsdb/chunks/head_chunks.go

```diff
@@ -376,7 +376,29 @@ func repairLastChunkFile(files map[int]string) (_ map[int]string, returnErr erro
 	if err != nil {
 		return files, errors.Wrap(err, "file stat during last head chunk file repair")
 	}
-	if info.Size() == 0 {
+
+	remove := info.Size() == 0 // Empty file.
+	if info.Size() >= MagicChunksSize {
+		// Check if magic number is 0.
+		f, err := os.Open(files[lastFile])
+		if err != nil {
+			return files, errors.Wrap(err, "open file during last head chunk file repair")
+		}
+
+		buf := make([]byte, MagicChunksSize)
+		_, err = f.Read(buf)
+		if err != nil {
+			return files, errors.Wrap(err, "read file during last head chunk file repair")
+		}
+
+		remove = binary.BigEndian.Uint32(buf) == 0
```

Comment:
I think we should log something here to tell the user that "hey, we encountered magic number is 0 error".

Reply:
Hmm, I don't think adding this log line is useful or actionable. Also, this package does not log anywhere and I would like to keep it that way. Deleting the last file is not a big issue.

Reply:
Yea, if we don't want to do fsync I don't think the log will be useful too.